src/test/java/com/artipie/rpm/pkg/ModifiableMetadataTest.java

```diff
@@ -95,6 +99,105 @@ void generatesMetadataFile(@TempDir final Path temp) throws IOException {
         );
     }
 
+    @Test
+    void generatesMetadataFileWhenRpmsWereRemoved(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("other.xml");
+        res.toFile().createNewFile();
+        final Path part = temp.resolve("part.other.xml");
+        Files.copy(Paths.get("src/test/resources-binary/repodata/other.xml.example"), part);
+        final ModifiableMetadata mtd = new ModifiableMetadata(
+            new MetadataFile(XmlPackage.OTHER, new OthersOutput(res).start()),
+            this.preceding(Optional.of(part))
+        );
+        mtd.close();
+        mtd.brush(
+            new ListOf<String>("54f1d9a1114fa85cd748174c57986004857b800fe9545fbf23af53f4791b31e2")
+        );
+        MatcherAssert.assertThat(
+            "Has 'aom' package, writes `packages` attribute correctly",
+            new XMLDocument(res),
+            Matchers.allOf(
+                XhtmlMatchers.hasXPath(
+                    "/*[local-name()='otherdata']/*[local-name()='package' and @name='aom']"
+                ),
+                new NodeHasPkgCount(1, XmlPackage.OTHER.tag())
+            )
+        );
+        MatcherAssert.assertThat(
+            "Does not have 'nginx' package",
+            new String(Files.readAllBytes(res), StandardCharsets.UTF_8),
+            new IsNot<>(
+                XhtmlMatchers.hasXPath(
+                    "/*[local-name()='otherdata']/*[local-name()='package' and @name='nginx']"
+                )
+            )
+        );
+    }
+
+    @Test
+    void generatesMetadataFileWhenRpmsWereAdded(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("filelists.xml");
+        res.toFile().createNewFile();
+        final Path part = temp.resolve("part.filelists.xml");
+        Files.copy(Paths.get("src/test/resources-binary/repodata/filelists.xml.example"), part);
```

Comment:
@olenagerasimova `TestResource` might be used here and in `generatesMetadataFileWhenRpmsWereRemoved` other test as it has been merged